### ISSUE TITLE
Another possible bug in `check_cell_boundaries_interval` at 0E

### ISSUE THREAD

**Issue by mo-gill** [ioos/compliance-checker#1298]
**Version of compliance checker running:*
1.7

**Describe the checker this affects:**
IOOS checker `check_cell_boundaries_interval`

**Attach a minimal CDL or NetCDF file which is able to reproduce the issue**
https://data.ceda.ac.uk/badc/cmip6/data/CMIP6/CMIP/MOHC/UKESM1-0-LL/piControl/r1i1p1f2/Amon/uas/gn/v20190410/uas_Amon_UKESM1-0-LL_piControl_r1i1p1f2_gn_196001-204912.nc

**To Reproduce:**
Run compliance-checker on the above file using `v1.7`,  whilst ensuring you have `netCDF4` version `1.7.4.` in the same environment.

**Describe the issue below:**
When checking cell bounds, the cell crossing 0E is being flagged as erroneous due to a misunderstanding around the bounds within the code. 

```
--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                 Version 5.4.2                                  
                     Report generated 2026-04-23T13:34:23Z                      
                                     cf:1.7                                     
http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html
--------------------------------------------------------------------------------
                               Corrective Actions                               
uas_Amon_UKESM1-0-LL_piControl_r1i1p1f2_gn_196001-204912.nc has 1 potential issue


                                    Warnings                                    
--------------------------------------------------------------------------------
§7.1 Cell Boundaries
* The points specified by the coordinate variable lon (0.0) lie outside the boundary of the cell specified by the associated boundary variable lon_bnds ([-0.9375  0.9375])
* The points specified by the coordinate variable lon (0.0) lie outside the boundary of the cell specified by the associated boundary variable lon_bnds ([-0.9375  0.9375])
* The points specified by the coordinate variable lon (0.0) lie outside the boundary of the cell specified by the associated boundary variable lon_bnds ([-0.9375  0.9375])
* The points specified by the coordinate variable lon (0.0) lie outside the boundary of the cell specified by the associated boundary variable lon_bnds ([-0.9375  0.9375])
* The points specified by the coordinate variable lon (0.0) lie outside the boundary of the cell specified by the associated boundary variable lon_bnds ([-0.9375  0.9375])
* The points specified by the coordinate variable lon (0.0) lie outside the boundary of the cell specified by the associated boundary variable lon_bnds ([-0.9375  0.9375])
```

**Comment by ocefpaf** [comment]
Indeed the way it is designed it can't correctly evaluate that:

```python
            for ii in range(len(variable[:])):
                if abs(boundary_variable[ii][1]) >= abs(boundary_variable[ii][0]):
                    if not ((abs(variable[ii]) >= abs(boundary_variable[ii][0])) and (abs(variable[ii]) <= abs(boundary_variable[ii][1]))):
                        valid = False
                        reasoning.append(
                            f"The points specified by the coordinate variable {variable_name} ({variable[ii]})"
                            " lie outside the boundary of the cell specified by the "
                            f"associated boundary variable {boundary_variable_name} ({boundary_variable[ii]})",
                        )
```

I'm not sure the implementation uses `abs` there. @benjwadams maybe you recall who implemented that and what was the logic at the time.

Edit: We could try

```python
if not boundary_variable[0] <= variable <= boundary_variable[1]
```

instead. I'll see if that check all the boxes.

**Comment by ocefpaf** [comment]
Actually. I can't reproduce that.

> Run compliance-checker on the above file using v1.7

You mean CF:1.7, but the compliance-checker used is 5.4.2 , correct? Are you specifying any extra checks? Can you paste the full environment and the command you are using to trigger this?

**Comment by mo-gill** [comment]
> Actually. I can't reproduce that.
> 
> > Run compliance-checker on the above file using v1.7
> 
> You mean CF:1.7, but the compliance-checker used is 5.4.2 , correct? Are you specifying any extra checks? Can you paste the full environment and the command you are using to trigger this?

You're right, i created a fresh environment and it works. 

I realised what was causing the errors in this issue was actually the "fix" i still had in my environment that worked for issue https://github.com/ioos/compliance-checker/issues/1296 , so perhaps reducing the indentation by one on line [380](https://github.com/ioos/compliance-checker/blob/main/compliance_checker/cf/cf_1_7.py#L380) might not be the right fix after all, despite it resolving the problem in #1296.

**Comment by ocefpaf** [comment]
Thanks, that makes sense with some of my findings. Let's close this one so we can focus on #1296 and #1299.
